src/net_processing.cpp

```diff
@@ -3557,32 +3557,49 @@ void ProcessMessage(
     return;
 }
 
+/** Maybe disconnect a peer and discourage future connections from its address.
+ *
+ * @param[in]   pnode     The node to check.
+ * @return                True if the peer was marked for disconnection in this function
+ */
 bool PeerLogicValidation::MaybeDiscourageAndDisconnect(CNode& pnode)
 {
-    LOCK(cs_main);
-    CNodeState &state = *State(pnode.GetId());
+    NodeId peer_id{pnode.GetId()};
+    {
+        LOCK(cs_main);
+        CNodeState &state = *State(peer_id);
```

Comment:
clang-format style-nit in commit 655b1957470c39bcab64917747c9f467444bd809

```suggestion
        CNodeState& state = *State(peer_id);
```

Reply:
Done in a follow-up: https://github.com/jnewbery/bitcoin/commit/5a3497f4e8b0d5ce00ecce8779e130b1c4c8ec0c